src/aioswitcher/api/tok.py

```diff
@@ -0,0 +1,36 @@
+# Copyright Tomer Figenblat.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# flake8: noqa
+
+"""Switcher integration TCP token handle module."""
+
+import base64
+import codecs
+
+magic = "ZnJvbSBDcnlwdG8uQ2lwaGVyIGltcG9ydCBBRVMNCmZyb20gQ3J5cHRvLlV0aWwuUGFkZGluZyBpbXBvcnQgdW5wYWQNCmZyb20gYmFzZTY0IGltcG9ydCBiNjRkZWNvZGUNCmZyb20gYmluYXNjaWkgaW1wb3J0IGh"
```

Comment:
these should be uppercase if they are constants

Reply:
@thecode If I do that it breaks the logic and it makes an error. Again this is a Black Box for us..

Reply:
Please finish committing your changes first (I noticed few places you wrote you will commit later or done and they are missing).

I will translate this code back to a normal Python code, this is a bad code obfuscation (as anyone can revert back to the original code) and I don't really understand the logic behind trying to obfuscate it as anyone can revert it back to the original code and it does nothing besides translating the token we get from switcher to be used in the lib so I don't understand the reason behind this step.

On the other hand this is like putting a binary code inside a Python lib which is very dangerous, how do you sure this code doesn't do something it shouldn't do?

Reply:
> Please finish committing your changes first (I noticed few places you wrote you will commit later or done and they are missing).
> 
> I will translate this code back to a normal Python code, this is a bad code obfuscation (as anyone can revert back to the original code) and I don't understand the logic behind trying to obfuscate it as anyone can revert it back to the original code and it does nothing besides translating the token we get from switcher to be used in the lib so I don't understand the reason behind this step.
> 
> On the other hand this is like putting a binary code inside a Python lib which is very dangerous, how do you sure this code doesn't do something it shouldn't do?

I am done with all the changes that are needed (you can see I commented on all the request changes stuff), I am just thinking of moving the token 2 functions logic (is_valid_token and convert the token to packet) to be part of DeviceToken.. I will do that later maybe.

If you can help with tok.py to convert that I will be more than happy.
I don't know how to do that, I got it as it is from Switcher devs, and proved to be working.. For me, it's like including a Python package I don't know why we are investing that much time on it.

This token is only used internally inside our house and is just another layer of safety.
This tok.py converts a "token" from Switcher based on the user's email into a packet string that should be part of the whole packet sent to a device. That is all it does.

Reply:
I have reversed the this code to its original code, it is only few lines of code and very simple, it is doing some imports during run which is not recommended. **Before I put this code back here, can you verify why Switcher try to hide this code**? I don't see any point in it as everyone can do the same as I did.
I would prefer that the original code will be here and not something which runs from binary which can easily break and someone might use it in the future to inject something (I don't want to explain here how but it is very simple).

Reply:
> I have reversed the this code to its original code, it is only few lines of code and very simple, it is doing some imports during run which is not recommended. **Before I put this code back here, can you verify why Switcher try to hide this code**? I don't see any point in it as everyone can do the same as I did. I would prefer that the original code will be here and not something which runs from binary which can easily break and someone might use it in the future to inject something (I don't want to explain here how but it is very simple).

It's just to make things harder for a potential hacker who already has access to our home network.. You can see what they did for the Homebridge Switcher plugin which uses the same logic: https://github.com/nitaybz/switcher-js/blob/master/src/tok.js

If you come up with a better code that does the same, please go for it and commit that..
I know we need that logic to get the packet string representing the token and that is it.

Reply:
What exactly does it make harder? the code is a function that has an input and produce and output? it is already on GitHub now, everyone can take this function and use it, it doesn't really matter if it is written so that it is not readable or not.